### ISSUE TITLE
Conversion to map[interface{}]interface{}

### ISSUE THREAD

**Issue by xor-gate** [vmihailenco/msgpack#200]
Hi @vmihailenco,

I have a value in JSON: `{"params":{"value":{"foo":"bar"}}`.  When I unmarshal it in a struct

```
type MyStruct struct {
       Params map[string]interface{}
}
```

The value of the value property is unmarshalled into `map[interface{}]interface{}`. As I only want to exchange data between JSON and messagepack I dont need the number keys from messagepack. So how I can force the package not to unmarshal into this map type but just a normal `map[string]interface{}` without specifing beforehand.

Hope this is clear.

Thanks in advance,
Jerry Jacobs

**Comment by vmihailenco** [comment]
Hi,

Latest version of this lib uses `map[string]interface{}` as new default. So you just need to upgate.

There is also [SetDecodeMapFunc](https://godoc.org/github.com/vmihailenco/msgpack#example-Decoder-SetDecodeMapFunc) which gives you full control over map decoding.

**Comment by xor-gate** [comment]
Yes we are currently using v2 API. I have tried v4 (latest master) it seems to crash but I will open a issue/PR when I figure out. This is the offending line: https://github.com/vmihailenco/msgpack/blob/b5e691b1eb52a28c05e67ab9df303626c095c23b/types.go#L140. When type is not a struct it panics here https://golang.org/src/reflect/value.go#L1191. It think it is caused because we have an aliased struct type to prevent recursion into itself.
